src/Test/L0/Worker/TestResults/XUnitResultReaderTests.cs

```diff
@@ -0,0 +1,400 @@
+        "<trait name=\"owner\" value=\"asdf\" />" +
+        "</traits>" +
+        "</test>" +
+        "</collection>" +
+        "</assembly>" +
+        "<assembly name=\"C:\\Users\\kaadhina\\Source\\Workspaces\\p1\\ClassLibrary2\\ClassLibrary1\\bin\\Debug\\ClassLibrary1.DLL\" environment=\"64-bit .NET 4.0.30319.42000 [collection-per-class, parallel]\" test-framework=\"xUnit.net 2.0.0.2929\" run-date=\"2015-08-18\" run-time=\"06:17:16\" config-file=\"C:\\Users\\kaadhina\\Source\\Workspaces\\p1\\ClassLibrary2\\packages\\xunit.runner.console.2.0.0\\tools\\xunit.console.exe.Config\" total=\"2\" passed=\"2\" failed=\"0\" skipped=\"0\" time=\"0.152\" errors=\"0\">" +
+        "<errors />" +
+        "<collection total=\"2\" passed=\"2\" failed=\"0\" skipped=\"0\" name=\"Test collection for MyFirstUnitTests.Class2\" time=\"0.006\">" +
+        "<test name=\"MyFirstUnitTests.Class2.tset2\" type=\"MyFirstUnitTests.Class2\" method=\"tset2\" time=\"0.0056931\" result=\"Pass\" />" +
+        "<test name=\"MyFirstUnitTests.Class2.test1\" type=\"MyFirstUnitTests.Class2\" method=\"test1\" time=\"0.0001093\" result=\"Pass\">" +
+        "<traits>" +
+        "<trait name=\"priority\" value=\"0\" />" +
+        "</traits>" +
+        "</test>" +
+        "</collection>" +
+        "</assembly>" +
+        "</assemblies>";
+
+        private const string _xunitResults = "<?xml version=\"1.0\" encoding=\"utf-8\"?>" +
+        "<assemblies>" +
+        "<assembly name = \"C:\\Users\\kaadhina\\Source\\Workspaces\\p1\\ClassLibrary2\\ClassLibrary2\\bin\\Debug\\ClassLibrary2.DLL\">" +
+        "<class name=\"MyFirstUnitTests.Class1\">" +
+        "<test name=\"MyFirstUnitTests.Class1.FailingTest\">" +
+        "</test>" +
+        "</class>" +
+        "</assembly>" +
+        "</assemblies>";
+
+        [Fact]
+        [Trait("Level", "L0")]
+        public void ResultsWithoutMandatoryFieldsAreSkipped()
+        {
+            SetupMocks();
+            string xunitResults = "<?xml version=\"1.0\" encoding=\"utf-8\"?>" +
+                                  "<assemblies>" +
+                                  "<assembly>" +
+                                  "<collection>" +
+                                  "<test name=\"\">" +
+                                  "</test>" +
+                                  "<test>" +
+                                  "</test>" +
+                                  "</collection>" +
+                                  "</assembly>" +
+                                  "</assemblies>";
+            WriteXUnitFile(xunitResults);
+            XUnitResultReader reader = new XUnitResultReader();
+            TestRunData runData = reader.ReadResults(_ec.Object, _xUnitResultFile, new TestRunContext("Owner", "any cpu", "debug", 1, "", "releaseUri", "releaseEnvironmentUri"));
+
+            Assert.NotNull(runData);
+            Assert.Equal(0, runData.Results.Length);
+        }
+
+        [Fact]
+        [Trait("Level", "L0")]
```

Comment:
Add another Trait to add category
[Trait("Category", "PublishTests")] or something better.. :)